daemon/algod/api/server/v2/handlers.go

```diff
@@ -732,6 +732,54 @@ func (v2 *Handlers) GetBlockTxids(ctx echo.Context, round uint64) error {
 	return ctx.JSON(http.StatusOK, response)
 }
 
+func getLogsFromTxns(txns []transactions.SignedTxnWithAD, blockLogs *[]model.BlockLog, outerTxnID string) {
+
+	for _, txn := range txns {
+		byteLogs := make([][]byte, len(txn.EvalDelta.Logs))
+		for i, log := range txn.EvalDelta.Logs {
+			byteLogs[i] = []byte(log)
+		}
+
+		*blockLogs = append(*blockLogs, model.BlockLog{
+			Txid:             outerTxnID,
+			Logs:             byteLogs,
+			ApplicationIndex: uint64(txn.ApplicationID),
```

Comment:
Two points I want to make here:
* Go embedding trickery: you might think this resolves to `txn.SignedTxn.Txn.ApplicationID`, but it's actually resolves to `txn.ApplyData.ApplicationID`. The `ApplicationID` name is duplicated across two different embedded structs, and for some reason (maybe definition order) go decides to resolve the `ApplyData` one. Best to be explicit in this case.
* In case this is an app create call (i.e. `txn.SignedTxn.Txn.ApplicationID == 0`), you _do_ want to actually reference `txn.ApplyData.ApplicationID`, as that will have the created app ID.

Reply:
Yes

Reply:
Resolved in [0d8e597](https://github.com/algorand/go-algorand/pull/5865/commits/0d8e597107444f3e8218246e77b23ee5572ecccf)